### PR TITLE
Mint Rewards

### PR DESCRIPTION
**Pull request by GWSzeto**

### CODE REVIEW

File: src/core/SplitFeesCore.sol [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    address public splitWalletImplementation;
+
+    /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(address _owner) {
+        _initializeOwner(_owner);
+
+        SplitWalletModule splitWalletModule = new SplitWalletModule();
```
Comment (GWSzeto) [inline src/core/SplitFeesCore.sol:59]: Discuss about the alternative options here

File: src/interface/IERC20.sol [hunk 2]
```diff
@@ -18,4 +18,10 @@ interface IERC20 {
 
     function transferFrom(address from, address to, uint256 value) external returns (bool);
 
+    function name() external view returns (string memory);
```
Comment (GWSzeto) [inline src/interface/IERC20.sol:21]: use the IERC20Metadata interface instead

File: src/module/SplitFeesModule.sol [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    function _setSplits(address[] memory _recipients, uint256[] memory _allocations, address _controller)
+        internal
+        pure
+        returns (Split memory)
+    {
+        Split memory _split;
+        uint256 _totalAllocation;
+        _split.recipients = _recipients;
+        _split.allocations = _allocations;
+        _split.totalAllocation = _totalAllocation;
```
Comment (GWSzeto) [inline src/module/SplitFeesModule.sol:186]: move this below the for loop

File: src/core/SplitFeesCore.sol [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        address[] memory modules = new address[](1);
+        bytes[] memory moduleInstallData = new bytes[](1);
+        modules[0] = address(splitWalletModule);
+
+        SplitWalletCore splitWalletCore = new SplitWalletCore(_owner, modules, moduleInstallData);
+        splitWalletImplementation = address(splitWalletCore);
+    }
```
Comment (GWSzeto) [inline src/core/SplitFeesCore.sol:67]: Discuss about the alternative implementations here

**Review by kumaryash90 — APPROVED** [review]

**Pull request merged**